File: gitdot-web/packages/api/src/endpoint/review/judge-review-diff.ts
import { z } from "zod";
import { ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const JudgeReviewDiffRequest = z.object({
  verdict: z.enum(["approve", "reject"]),
});
export type JudgeReviewDiffRequest = z.infer<typeof JudgeReviewDiffRequest>;

export const JudgeReviewDiffResponse = ReviewResource;
export type JudgeReviewDiffResponse = z.infer<typeof JudgeReviewDiffResponse>;

export const JudgeReviewDiff = {
  path: "/repository/{owner}/{repo}/review/{number}/diff/{position}/submit",
  method: "POST",
  request: JudgeReviewDiffRequest,
  response: JudgeReviewDiffResponse,
} as const satisfies Endpoint;
export type JudgeReviewDiff = typeof JudgeReviewDiff;

File: gitdot-api/src/endpoint/review/judge_review_diff.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::review::ReviewResource};

pub struct JudgeReviewDiff;

impl Endpoint for JudgeReviewDiff {
    const PATH: &'static str = "/repository/{owner}/{repo}/review/{number}/diff/{position}/submit";
    const METHOD: http::Method = http::Method::POST;

    type Request = JudgeReviewDiffRequest;
    type Response = JudgeReviewDiffResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct JudgeReviewDiffRequest {
    pub verdict: String,
}

pub type JudgeReviewDiffResponse = ReviewResource;
